lib/httpRequestBuilder.js

```diff
@@ -13,12 +13,63 @@ function requestBuilder (defaults) {
     headers: {},
     body: Buffer.alloc(0),
     hostname: 'localhost',
+    chunkSize: 0,
     setupRequest: reqData => reqData,
     port: 80
   }
 
   defaults = Object.assign(builderDefaults, defaults)
 
+  // cache chunks to avoid reprocessing
+  const chunkBodyCache = {}
```

Comment:
This might leak memory if the body is changed. I would recommend using an lru cache, such as the one provided in the tiny-lru package.